source/secure_tunneling.c

```diff
@@ -14,6 +14,54 @@
 
 #define UNUSED(x) (void)(x)
 
+static struct aws_array_list s_active_tunnels; // A list of `struct aws_secure_tunnel *`.
+static bool s_active_tunnels_initialized = false;
+
+static void s_active_tunnels_initialize(struct aws_allocator *alloc) {
```

Comment:
Prefer to initialize libraries via explicit library_init() function calls that create all necessary internal structures over lazy initialization of specific internal structure.